student-os/scripts/materials_convert.py

```
@@ -515,20 +518,23 @@ def http_json(method: str, url: str, *, payload: dict[str, Any] | None = None, t
-    except urllib.error.URLError as exc:
-        raise RuntimeError(f"MinerU v1 Agent upload failed: {exc.reason}") from exc
+        connection.request("PUT", path, body=data, headers={"Content-Length": str(len(data))})
+        response = connection.getresponse()
+        response_body = response.read()
+    except OSError as exc:
+        raise RuntimeError(f"MinerU v1 Agent upload failed: {exc}") from exc
+    finally:
+        connection.close()
+    if response.status >= 400:
+        detail = response_body.decode("utf-8", errors="replace").strip()
+        raise RuntimeError(f"MinerU v1 Agent upload HTTP {response.status}: {detail}") from None
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)